package/ContextMenu.tsx

```diff
@@ -41,11 +43,13 @@ export function ContextMenu({
   sx,
   classNames,
   styles,
+  subOptions,
 }: ContextMenuProps) {
   useWindowEvent('resize', onHide);
   useWindowEvent('scroll', onHide);
 
   const clickOutsideRef = useClickOutside<HTMLDivElement>(onHide);
```

Comment:
Last thing I'm struggling with is the `useClickOutside` because when I click on the submenu (another instance of `<ContextMenu/>`, it considers to be outside of the main one, and therefore closes it before it register the click on the submenu. @icflorescu Do you have any idea how I could make the clickOutside handler work on multiple `<ContextMenu/>` at the same time? 

I tried the [Mantine `useClickOutside`'s `nodes`](https://mantine.dev/hooks/use-click-outside/#multiple-nodes) argument but unfortunately it cannot take a real ref (useRef), only useState based refs, which don't work with `useMergedRef` which we need on line 55.

Reply:
Thanks a lot for your work, @Corvimia!

Sorry for being offline for the last couple of weeks; I still have to deal with some personal issues before being able to really get back to this.

The issue with `useClickOutside` is indeed a bummer.
TBH, I would have taken the exact same approach as you did, but obviously it's not working on multiple, nested context menus.

Just give me a few more days and I'll try to think of a different approach to see how we could bypass the issue.